### ISSUE TITLE
rsync fails with error code 11 during image unpack (unpackfs) if using LUKS2, works with LUKS1

### ISSUE THREAD

**Issue by GloriousEggroll** [calamares/calamares#2108]
Configurations used:

$ grep -v '^\s*$\|^\s*\#' partition.conf 
```
efiSystemPartition:     "/boot/efi"
efiSystemPartitionSize:     600M
userSwapChoices:
    - none      # Create no swap, use no swap
    - small     # Up to 4GB
    - suspend   # At least main memory size
    - file      # To swap file instead of partition
luksGeneration: luks2
drawNestedPartitions:   true
alwaysShowPartitionLabels: true
initialPartitioningChoice: none
initialSwapChoice: file
defaultFileSystemType:  "ext4"
```

$ grep -v '^\s*$\|^\s*\#' ../settings.conf 
```
---
modules-search: [ local ]
sequence:
- show:
  - welcome
  - locale
  - keyboard
  - partition
  - users
  - summary
- exec:
  - partition
  - mount
  - unpackfs
  - machineid
  - fstab
  - locale
  - keyboard
  - localecfg
  - luksbootkeyfile
  - dracutlukscfg
  - users
  - displaymanager
  - networkcfg
  - hwclock
  - services-systemd
  - dracut
  - grubcfg
  - bootloader
  - packages
  - shellprocess
  - umount
- show:
  - finished
branding: auto
prompt-install: false
dont-chroot: false
oem-setup: false
disable-cancel: false
disable-cancel-during-exec: false
hide-back-and-next-during-exec: false
quit-at-end: false
```

GUI steps:
Choose 'Erase Disk', 'Swap to file'. Check 'Encrypt system' and enter a password.
You should end up with 'EFI System': Fat32 and '*** Linux': LUKS2 in the 'After' section.

Click next, fill in user data. Click next, press install.

Result:
![Screenshot from 2023-03-03 12-14-00](https://user-images.githubusercontent.com/11287837/222806966-7d41269f-0367-422b-9948-cfbe1a90f482.png)

Debug log captured like so:
$ sudo -E calamares -d &> calamares_debug.log

Debug log attached.
[calamares_debug.log](https://github.com/calamares/calamares/files/10885431/calamares_debug.log)

**Comment by GloriousEggroll** [comment]
Additional notes:

-Using Nobara 37, base is Fedora 37
-Build options:
```
sed -i -e "s|CALAMARES_VERSION_RC 1|CALAMARES_VERSION_RC 0|g" CMakeLists.txt

%{cmake_kf5} -DCMAKE_BUILD_TYPE:STRING="RelWithDebInfo" \
             -DBUILD_TESTING:BOOL=OFF \
             -DWITH_PYTHONQT:BOOL=OFF \
             -DWITH_KPMCORE4API:BOOL=ON
```

**Comment by dalto8** [comment]
It is failing to mount.

```
    .. Target cmd: ("mount", "-t", "luks2", "-o", "defaults,noatime", "/dev/mapper/luks-7613dbe4-2cfd-41e4-ba6c-fae7401b0007", "/tmp/calamares-root-syzerqnr/") Exit code: 32 output:
 mount: /tmp/calamares-root-syzerqnr: unknown filesystem type 'luks2'.
```

Are you building straight off the `calamares` branch?

**Comment by GloriousEggroll** [comment]
> It is failing to mount.
> 
> ```
>     .. Target cmd: ("mount", "-t", "luks2", "-o", "defaults,noatime", "/dev/mapper/luks-7613dbe4-2cfd-41e4-ba6c-fae7401b0007", "/tmp/calamares-root-syzerqnr/") Exit code: 32 output:
>  mount: /tmp/calamares-root-syzerqnr: unknown filesystem type 'luks2'.
> ```
> 
> Are you building straight off the `calamares` branch?

I am building off the calamares branch, but the version tag shows alpha:

14:08:50 [6]: void CalamaresApplication::init()
    Calamares version: 3.3.0-alpha3

**Comment by dalto8** [comment]
Possibly the recent partition module changes aren't compatible with the mount module.  I will take a look at it tonight or tomorrow unless someone else gets to it first.

**Comment by adriaandegroot** [comment]
There's some leftover kpmcore-multiple-api-compatibility code that is now wrong (because every supported kpmcore version is the 4.2 API or later, we no longer check which one it is).

**Comment by dalto8** [comment]
Should be fixed with f5b974984cc15da8486c15f9862a49df40072351.

Let me know if you are still having trouble.

**Comment by GloriousEggroll** [comment]
can confirm it works perfectly now, thank you!
